app/src/main/java/com/github/wnder/guessLocation/GuessFragment.java

```diff
@@ -0,0 +1,243 @@
+
+    @Inject
+    public NetworkService networkInfo;
+
+    private TextView radiusTextView;
+    private MapView mapView;
+    private MapboxMap mapboxMap;
+    private LatLng cameraPosition;
+    private User user;
+
+    public GuessFragment() {}
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                              Bundle savedInstanceState){
+        Mapbox.getInstance(getActivity(), getString(R.string.mapbox_access_token));
+        View rootView = inflater.inflate(R.layout.fragment_seekbar, container, false);
+
+        user = GlobalUser.getUser();
+
+        mapView = rootView.findViewById(R.id.mapViewFragment);
+        mapView.onCreate(savedInstanceState);
+        mapView.getMapAsync(this);
+
+        double cameraLat = user.getPositionFromGPS((LocationManager) getActivity().getSystemService(Context.LOCATION_SERVICE), getActivity()).getLatitude();
+        double cameraLng = user.getPositionFromGPS((LocationManager) getActivity().getSystemService(Context.LOCATION_SERVICE), getActivity()).getLongitude();
+        cameraPosition = new LatLng(cameraLat, cameraLng);
+
+        SeekBar radiusSeekBar = rootView.findViewById(R.id.radiusSeekBar);
+        radiusTextView = rootView.findViewById(R.id.radiusTextView);
+        manageSeekBar(radiusSeekBar, radiusTextView);
+
+        FloatingActionButton guessButton = rootView.findViewById(R.id.navigationToGuessButton);
+        guessButton.setOnClickListener((view) -> openPreviewActivity());
+
+        return rootView;
+    }
+
+    private void setZoom(){
+        //Set camera position
+        CameraPosition position = new CameraPosition.Builder()
+                .target(cameraPosition)
+                .zoom(zoomFromKilometers(cameraPosition, user.getRadius()))
+                .build();
+        mapboxMap.setCameraPosition(position);
+    }
+
+
+    @Override
+    public void onMapReady(@NonNull MapboxMap mapboxMap) {
+        this.mapboxMap = mapboxMap;
+        setZoom();
+
+        //Set mapbox style
+        mapboxMap.getUiSettings().setCompassEnabled(false); //Hide the default mapbox compass because we use our compass
+
+        mapboxMap.setStyle(Style.SATELLITE_STREETS, this::onStyleLoaded);
+    }
+
+    /**
+     * To be executed when the style has loaded
+     * @param style on the mapbox map
+     */
+    private void onStyleLoaded(Style style){
+        drawCircle(getActivity(), mapboxMap, cameraPosition);
+    }
+
+    private void openPreviewActivity() {
+        if(networkInfo.isNetworkAvailable()){
+            Intent intent = new Intent(getActivity(), GuessPreviewActivity.class);
+            startActivity(intent);
+        }
+        else{
+            AlertDialog alert = AlertBuilder.okAlert(getString(R.string.no_connection), getString(R.string.no_internet_guess), getActivity());
+            alert.show();
+        }
+    }
+
+    //Necessary overwrites for MapView lifecycle methods
+    /**
```

Comment:
Just for clarity, I would move all the overrides that don't do anything to the end of the file an put actual code like `onProgressChanged` up there.

Reply:
done